node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -411,6 +404,26 @@ func (atp *apiTransactionProcessor) extractNonceGaps(sender string, senderShard
 	return nonceGaps, nil
 }
 
+func (atp *apiTransactionProcessor) appendGapFromAccountNonceIfNeeded(
+	senderAccountNonce uint64,
+	firstNonceInPool uint64,
+	senderShard uint32,
+	nonceGaps *[]common.NonceGapApiResponse,
+) {
+	if atp.shardCoordinator.SelfId() != senderShard {
+		return
+	}
+
+	nonceDif := firstNonceInPool - senderAccountNonce
+	if nonceDif > 1 {
```

Comment:
This doesn't cover all the situations. For example, considering the wallet of Alice with nonce = 5. In the pool, the smallest Alice's nonce is 6. Therefore, `6-5=1`, and `1` is not greater than `1`, so nothing will be appended. 

So it won't work for a single nonce gap.

Reply:
pushed